helsingborg-stad/helsingborg-io-sls-api
# feat: add status api for fetching api status message

## Explain the changes you’ve made
Added a new api (status-api) for fetching api status messages.

## Explain why these changes are made
In order to actively set the backend or application in maintenance we wanted a lambda function to return a status message which could be visible in the application. If the application receives a message from the endpoint, we sign out the user and shows the message.

## Explain your solution
Created a lambda function called `getStatus` which returns the message held in an environment variable. 
By having the value as an environment variable, we could easily change the value in the Lambda console and it would be fetched by the application immediately without re-deploying anything.
By having it as an environment variable we also skip the creation of a database.

## How to test

1. Checkout this branch
2. run `sls deploy` within service folder for `service status-api`
3. Make sure you have `status` gateway path deployed 
4. Run the application in `production` mode, make sure you have all changes needed from this PR: https://github.com/helsingborg-stad/app-mitt-helsingborg/pull/621
5. Go to the `getStatus` lambda function and change the message environment variable and check the result in the application.

## Tested environments

- [] Your local Serverless environment.
